docs/src/modules/java/pages/value-entity.adoc

```diff
@@ -235,6 +209,39 @@ By default, the generated constructor has a `ValueEntityContext` parameter, but
 
 When more components are added the `KalixFactory` is regenerated, and you have to adjust the registration from the `Main` class.
 
+=== Deleting state
+
+The next example shows how to delete a Value Entity state by returning special `deleteEntity()` effect.
+
+[.tabset]
+Java::
++
+[source,java,indent=0]
+./src/main/java/com/example/domain/Counter.java
+----
+include::java:example$java-valueentity-counter/src/main/java/com/example/domain/Counter.java[tag=delete]
+----
+<1> We delete the state by returning an `Effect` with `effects().deleteEntity()`.
+
+Scala::
++
+[source,scala,indent=0]
+./src/main/java/com/example/domain/Counter.scala
+----
+include::java:example$scala-valueentity-counter/src/main/scala/com/example/domain/Counter.scala[tag=delete]
+----
+<1> We delete the state by returning an `Effect` with `effects.deleteEntity()`.
+
+When you give the instruction to delete the entity it will still exist with an empty state from some time. The actual removal happens later to give downstream consumers time to process the change. By default, the existence of the entity is completely cleaned up after a week.
```

Comment:
```suggestion
When you give the instruction to delete the entity it will still exist with an empty state for some time. The actual removal happens later to give downstream consumers time to process the change. By default, the existence of the entity is completely cleaned up after a week.
```